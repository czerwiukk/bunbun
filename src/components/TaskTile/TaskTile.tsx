import { faGripVertical, faSpinner } from "@fortawesome/free-solid-svg-icons";
import { createDraggable } from "@thisbeyond/solid-dnd";
import Fa from "solid-fa";
import { Component } from "solid-js";
import { IBoardTask } from "~/types";

interface TaskTileProps {
  task: IBoardTask;
}

export const TaskTile: Component<TaskTileProps> = (props) => {
  const draggable = createDraggable(props.task.id);

  return (
    <li
      // @ts-ignore
      use:draggable
      class="flex flex-col relative h-24 mb-4 border border-stone-300 bg-stone-100 active:bg-stone-50 rounded-md text-stone-700 px-4 py-2 draggable cursor-grab active:cursor-grabbing active:z-30 active:shadow-lg overflow-hidden text-ellipsis"
    >
      <p class="font-bold mb-1 truncate">{props.task.name}</p>

      <p class="text-xs truncate">{props.task.description}</p>

      <div class="absolute right-2 bottom-2 text-stone-400">
        <Fa icon={faGripVertical} />
      </div>
    </li>
  );
};
