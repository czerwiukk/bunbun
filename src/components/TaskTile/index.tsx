import { faGripVertical, faSpinner } from "@fortawesome/free-solid-svg-icons";
import { createDraggable } from "@thisbeyond/solid-dnd";
import Fa from "solid-fa";
import { Component } from "solid-js";
import { IBoardTask } from "~/types";

interface TaskTileProps {
  task: IBoardTask;
}

export const TaskTile: Component<TaskTileProps> = (props) => {
  const draggable = createDraggable(props.task.id);

  return (
    <li
      // @ts-ignore
      use:draggable
      class="flex flex-col relative h-32 w-full bg-stone-100 rounded-md text-stone-700 px-4 py-2 draggable cursor-grab active:cursor-grabbing mb-4"
    >
      <p class="font-bold text-lg mb-1 cursor-text">{props.task.name}</p>

      <p class="text-sm cursor-text">{props.task.description}</p>

      <div class="absolute right-2 bottom-2 text-stone-400">
        <Fa icon={faGripVertical} />
      </div>
    </li>
  );
};
