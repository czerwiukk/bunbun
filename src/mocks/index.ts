import { IBoardColumn, IBoardTask } from "~/types";

export const tasks: IBoardTask[] = [
  {
    id: 1,
    name: "Test",
    description: "Testowy test",
    column_id: 1,
    created_at: "2021-08-01T12:00:00.000Z",
  },
  {
    id: 2,
    name: "Test2",
    description: "Testowy test2",
    column_id: 1,
    created_at: "2021-08-01T12:00:00.000Z",
  },
  {
    id: 3,
    name: "Test3",
    description: "Testowy test3",
    column_id: 1,
    created_at: "2021-08-01T12:00:00.000Z",
  },
];

export const columns: IBoardColumn[] = [
  {
    id: 1,
    name: "Column1",
    created_at: "2021-08-01T12:00:00.000Z",
  },
  {
    id: 2,
    name: "Column2",
    created_at: "2021-08-01T12:00:00.000Z",
  },
];
