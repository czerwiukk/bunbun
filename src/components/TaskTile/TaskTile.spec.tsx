import { render } from "solid-testing-library";
import { describe, expect, it, vi } from "vitest";
import { tasks } from "~/mocks";
import { TaskTile } from "./TaskTile";

vi.mock("@thisbeyond/solid-dnd", () => ({
  createDraggable: () => () => ({}),
}));

describe("TaskTile", () => {
  it("matches snapshot", () => {
    expect(render(() => <TaskTile task={tasks[0]} />, {})).toMatchSnapshot();
  });
});
