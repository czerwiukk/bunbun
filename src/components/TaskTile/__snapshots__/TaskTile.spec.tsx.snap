// Vitest Snapshot v1

exports[`TaskTile > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="flex flex-col relative h-24 mb-4 border border-stone-300 dark:border-stone-700 bg-stone-100 dark:bg-stone-900 active:bg-stone-50 dark:active:bg-stone-700 rounded-md text-stone-700 dark:text-stone-100 px-4 py-2 draggable cursor-grab active:cursor-grabbing active:z-30 active:shadow-lg overflow-hidden text-ellipsis"
      >
        <p
          class="font-bold mb-1 truncate"
        >
          Test
        </p>
        <p
          class="text-xs truncate"
        >
          Testowy test
        </p>
        <div
          class="absolute right-2 bottom-2 text-stone-400"
        >
          <svg
            aria-hidden="true"
            class="fa"
            style="height: 1em; vertical-align: -.125em; transform-origin: center; overflow: visible;"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              transform="translate(160 256)"
              transform-origin="80 0"
            >
              <g
                transform="translate(0,0) scale(1,1)"
              >
                <path
                  d="M40 352c-22.1 0-40 17.9-40 40l0 48c0 22.1 17.9 40 40 40l48 0c22.1 0 40-17.9 40-40l0-48c0-22.1-17.9-40-40-40l-48 0zm192 0c-22.1 0-40 17.9-40 40l0 48c0 22.1 17.9 40 40 40l48 0c22.1 0 40-17.9 40-40l0-48c0-22.1-17.9-40-40-40l-48 0zM40 320l48 0c22.1 0 40-17.9 40-40l0-48c0-22.1-17.9-40-40-40l-48 0c-22.1 0-40 17.9-40 40l0 48c0 22.1 17.9 40 40 40zM232 192c-22.1 0-40 17.9-40 40l0 48c0 22.1 17.9 40 40 40l48 0c22.1 0 40-17.9 40-40l0-48c0-22.1-17.9-40-40-40l-48 0zM40 160l48 0c22.1 0 40-17.9 40-40l0-48c0-22.1-17.9-40-40-40L40 32C17.9 32 0 49.9 0 72l0 48c0 22.1 17.9 40 40 40zM232 32c-22.1 0-40 17.9-40 40l0 48c0 22.1 17.9 40 40 40l48 0c22.1 0 40-17.9 40-40l0-48c0-22.1-17.9-40-40-40l-48 0z"
                  fill="currentColor"
                  transform="translate(-160 -256)"
                />
              </g>
            </g>
          </svg>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="flex flex-col relative h-24 mb-4 border border-stone-300 dark:border-stone-700 bg-stone-100 dark:bg-stone-900 active:bg-stone-50 dark:active:bg-stone-700 rounded-md text-stone-700 dark:text-stone-100 px-4 py-2 draggable cursor-grab active:cursor-grabbing active:z-30 active:shadow-lg overflow-hidden text-ellipsis"
    >
      <p
        class="font-bold mb-1 truncate"
      >
        Test
      </p>
      <p
        class="text-xs truncate"
      >
        Testowy test
      </p>
      <div
        class="absolute right-2 bottom-2 text-stone-400"
      >
        <svg
          aria-hidden="true"
          class="fa"
          style="height: 1em; vertical-align: -.125em; transform-origin: center; overflow: visible;"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(160 256)"
            transform-origin="80 0"
          >
            <g
              transform="translate(0,0) scale(1,1)"
            >
              <path
                d="M40 352c-22.1 0-40 17.9-40 40l0 48c0 22.1 17.9 40 40 40l48 0c22.1 0 40-17.9 40-40l0-48c0-22.1-17.9-40-40-40l-48 0zm192 0c-22.1 0-40 17.9-40 40l0 48c0 22.1 17.9 40 40 40l48 0c22.1 0 40-17.9 40-40l0-48c0-22.1-17.9-40-40-40l-48 0zM40 320l48 0c22.1 0 40-17.9 40-40l0-48c0-22.1-17.9-40-40-40l-48 0c-22.1 0-40 17.9-40 40l0 48c0 22.1 17.9 40 40 40zM232 192c-22.1 0-40 17.9-40 40l0 48c0 22.1 17.9 40 40 40l48 0c22.1 0 40-17.9 40-40l0-48c0-22.1-17.9-40-40-40l-48 0zM40 160l48 0c22.1 0 40-17.9 40-40l0-48c0-22.1-17.9-40-40-40L40 32C17.9 32 0 49.9 0 72l0 48c0 22.1 17.9 40 40 40zM232 32c-22.1 0-40 17.9-40 40l0 48c0 22.1 17.9 40 40 40l48 0c22.1 0 40-17.9 40-40l0-48c0-22.1-17.9-40-40-40l-48 0z"
                fill="currentColor"
                transform="translate(-160 -256)"
              />
            </g>
          </g>
        </svg>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "unmount": [Function],
}
`;
